packages/common/src/downloads/getDownloadFormats.ts

```diff
@@ -0,0 +1,76 @@
+import { IHubAdditionalResource } from "../core/types/IHubAdditionalResource";
+import { getProp } from "../objects/get-prop";
+import { canUseExportImageFlow } from "./_internal/canUseExportImageFlow";
+import { getExportImageFormats } from "./_internal/format-fetchers/getExportImageFormats";
+import { canUseHubDownloadApi } from "./canUseHubDownloadApi";
+import { getDownloadConfiguration } from "./getDownloadConfiguration";
+import { getHubDownloadApiFormats } from "./getHubDownloadApiFormats";
+import {
+  IDownloadFormat,
+  IDynamicDownloadFormat,
+  IFetchDownloadFormatsOptions,
+  IStaticDownloadFormat,
+} from "./types";
+
+/**
+ * Gets available download formats / additional resources for the given entity in the order they have been configured.
+ * If a format has been configured to be hidden, it will not be included in the results.
+ *
+ * @param options options to refine / filter the results of the fetchDownloadFormats operation
+ * @returns The available download formats and additional resources
+ */
+export function getDownloadFormats(
+  options: IFetchDownloadFormatsOptions
+): IDownloadFormat[] {
+  const { entity, context } = options;
+  // get the base formats for the item
+  let baseFormats: IDynamicDownloadFormat[] = [];
+  // TODO: use typescript to enforce a branch for each flow type
+  if (canUseHubDownloadApi(entity, context)) {
+    baseFormats = getHubDownloadApiFormats(entity);
+  } else if (canUseExportImageFlow(entity)) {
+    baseFormats = getExportImageFormats(entity);
+  }
+
+  // add additional resource links as static formats
+  const additionalResources =
+    getProp(entity, "extendedProps.additionalResources") || [];
+  const additionalFormats = additionalResources.map(toStaticFormat);
+
+  // Respect the order and visibility of the formats as configured for the entity
+  const downloadConfiguration = getDownloadConfiguration(entity);
+  return downloadConfiguration.formats.reduce((acc, format) => {
+    if (!format.hidden) {
+      let includedFormat;
+      // TODO: use utility function to determine if the format is an additional resource
+      const isAdditionalResource = format.key.startsWith(
```

Comment:
Don't we have a helper function for this in this PR?

Reply:
moved to private internal helper for reuse.